File: VB/MultipleORMsExample.Module/BusinessObjects/EntityFrameworkSampleObject.vb
Imports System.Linq
Imports System.Text
Imports System.ComponentModel
Imports DevExpress.Persistent.Base
Imports System.Data.Entity
Imports DevExpress.ExpressApp.DC

Namespace MultipleORMsExample.Module.BusinessObjects
    <DefaultClassOptions> _
    Public Class EntityFrameworkSampleObject
        Private privateId As Integer
        <Browsable(False)> _
        Public Property Id() As Integer
            Get
                Return privateId
            End Get
            Protected Set(ByVal value As Integer)
                privateId = value
            End Set
        End Property
        Private privateName As String
        Public Property Name() As String
            Get
                Return privateName
            End Get
            Set(ByVal value As String)
                privateName = value
            End Set
        End Property
        Private privateDescription As String
        <FieldSize(FieldSizeAttribute.Unlimited)> _
        Public Property Description() As String
            Get
                Return privateDescription
            End Get
            Set(ByVal value As String)
                privateDescription = value
            End Set
        End Property
    End Class

    Public Class MyDbContext
        Inherits DbContext

        Public Sub New(ByVal connectionString As String)
            MyBase.New(connectionString)
        End Sub
        Private privateSampleObjects As DbSet(Of EntityFrameworkSampleObject)
        Public Property SampleObjects() As DbSet(Of EntityFrameworkSampleObject)
            Get
                Return privateSampleObjects
            End Get
            Set(ByVal value As DbSet(Of EntityFrameworkSampleObject))
                privateSampleObjects = value
            End Set
        End Property
    End Class
End Namespace

File: CS/MultipleORMsExample.Module/BusinessObjects/EntityFrameworkSampleObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using DevExpress.Persistent.Base;
using System.Data.Entity;
using DevExpress.ExpressApp.DC;
using DevExpress.ExpressApp.Updating;

namespace MultipleORMsExample.Module.BusinessObjects {
    [DefaultClassOptions]
    public class EntityFrameworkSampleObject {
        [Browsable(false)]
        public int Id { get; protected set; }
        public string Name { get; set; }
        [FieldSize(FieldSizeAttribute.Unlimited)]
        public String Description { get; set; }
    }

    public class MyDbContext : DbContext {
        public MyDbContext(string connectionString) : base(connectionString) { }
        public DbSet<EntityFrameworkSampleObject> SampleObjects { get; set; }
        public DbSet<DevExpress.ExpressApp.EF.Updating.ModuleInfo> ModulesInfo { get; set; }
    }
}
